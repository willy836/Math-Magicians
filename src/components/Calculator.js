import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [calculatorObj, setCalculatorObj] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (e) => {
    const btnName = e.target.innerHTML;
    setCalculatorObj((currentState) => calculate(
      {
        total: currentState.total,
        next: currentState.next,
        operation: currentState.operation,
      },
      btnName,
    ));
  };

  const { total, next, operation } = calculatorObj;
  return (
    <>
      <div className="cal-homepage">
        <h2>Let&apos;s do some math!</h2>
        <div className="calculator">
          <div className="grid-item grid-item1">
            {total}
            {' '}
            {operation}
            {' '}
            {next}
          </div>
          <button
            type="button"
            name="AC"
            className="grid-item"
            onClick={handleClick}
          >
            AC
          </button>
          <button
            type="button"
            name="+/-"
            className="grid-item"
            onClick={handleClick}
          >
            +/-
          </button>
          <button
            type="button"
            name="%"
            className="grid-item"
            onClick={handleClick}
          >
            %
          </button>
          <button
            type="button"
            name="รท"
            className="grid-item orange-bg"
            onClick={handleClick}
          >
            รท
          </button>
          <button
            type="button"
            name="7"
            className="grid-item"
            onClick={handleClick}
          >
            7
          </button>
          <button
            type="button"
            name="8"
            className="grid-item"
            onClick={handleClick}
          >
            8
          </button>
          <button
            type="button"
            name="9"
            className="grid-item"
            onClick={handleClick}
          >
            9
          </button>
          <button
            type="button"
            name="x"
            className="grid-item orange-bg"
            onClick={handleClick}
          >
            x
          </button>
          <button
            type="button"
            name="4"
            className="grid-item"
            onClick={handleClick}
          >
            4
          </button>
          <button
            type="button"
            name="5"
            className="grid-item"
            onClick={handleClick}
          >
            5
          </button>
          <button
            type="button"
            name="6"
            className="grid-item"
            onClick={handleClick}
          >
            6
          </button>
          <button
            type="button"
            name="-"
            className="grid-item orange-bg"
            onClick={handleClick}
          >
            -
          </button>
          <button
            type="button"
            name="1"
            className="grid-item"
            onClick={handleClick}
          >
            1
          </button>
          <button
            type="button"
            name="2"
            className="grid-item"
            onClick={handleClick}
          >
            2
          </button>
          <button
            type="button"
            name="3"
            className="grid-item"
            onClick={handleClick}
          >
            3
          </button>
          <button
            type="button"
            name="+"
            className="grid-item orange-bg"
            onClick={handleClick}
          >
            +
          </button>
          <button
            type="button"
            name="0"
            className="grid-item grid-item0"
            onClick={handleClick}
          >
            0
          </button>
          <button
            type="button"
            name="."
            className="grid-item"
            onClick={handleClick}
          >
            .
          </button>
          <button
            type="button"
            name="="
            className="grid-item orange-bg"
            onClick={handleClick}
          >
            =
          </button>
        </div>
      </div>
    </>
  );
};

export default Calculator;
