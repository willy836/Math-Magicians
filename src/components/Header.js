import React from 'react';
import { Link } from 'react-router-dom';

const Header = () => (
  <div className="header">
    <nav>
      <h1>Math Magicians</h1>
      <div className="links">
        <Link to="/" style={{ color: 'blue', fontSize: 18 }}>
          Home
        </Link>
        |
        <Link to="/Calculator" style={{ color: 'blue', fontSize: 18 }}>
          Calculator
        </Link>
        |
        <Link to="/Quote" style={{ color: 'blue', fontSize: 18 }}>
          Quote
        </Link>
      </div>
    </nav>
  </div>
);

export default Header;
