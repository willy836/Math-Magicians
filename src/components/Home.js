import React from 'react';

const Home = () => (
  <div className="home">
    <h2 className="home-h2">Welcome to our page!</h2>
    <div className="home-content">
      <p className="home-para">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Magni fugiat
        veniam iure explicabo! Magni iusto fugit, quo illo vero eaque.
        Recusandae laboriosam deleniti provident sit qui quasi tenetur iure
        quam rerum maxime ipsum suscipit eligendi quae incidunt sint doloribus
        harum, illum error iusto natus ipsam! Ad corrupti dolore delectus
        praesentium.
      </p>

      <p className="home-para">
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur omnis
        odit vero iure reiciendis voluptas molestias numquam sequi? Deserunt
        voluptas accusantium labore laudantium consequuntur natus repudiandae
        sit quas quam aliquid fugiat commodi itaque, possimus deleniti eum
        modi delectus eligendi at quaerat ea cum ullam. Impedit iusto totam
        dolore pariatur harum doloribus corporis amet minima recusandae odit
        illo ipsum cum ratione atque, aliquid odio nesciunt consequuntur
        facere quos? Ea obcaecati neque placeat laborum aperiam, accusamus
        sunt excepturi veniam, adipisci qui distinctio.
      </p>
    </div>
  </div>
);

export default Home;
