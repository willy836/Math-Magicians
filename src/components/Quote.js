import React from 'react';

const Quote = () => (
  <div className="random-q">
    <p className="quote">
      Mathematics is not about numbers, equations, computations, or
      algorithms: it is about understanding. - William Paul Thurston
    </p>
  </div>
);

export default Quote;
