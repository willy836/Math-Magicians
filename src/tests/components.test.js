import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import Header from '../components/Header';
import Quote from '../components/Quote';
import Home from '../components/Home';
import Calculator from '../components/Calculator';

describe('Home and Quote components render correctly', () => {
  it('Home component should match the snapshot', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Quote component should match the snapshot', () => {
    const tree = renderer.create(<Quote />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// Test links in Header component
const MockHeaderLinks = () => {
  <BrowserRouter>
    <Header />
  </BrowserRouter>;
};
describe('Test links in Header component', () => {
  it('Header links render correctly', () => {
    const tree = renderer.create(<MockHeaderLinks />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// Test Calculator component
describe('Test Calculator component', () => {
  it('Test correct render of user interactions with calculator', () => {
    render(<Calculator />);
    fireEvent.click(screen.getByRole('button', { name: '2' }));
    fireEvent.click(screen.getByRole('button', { name: '0' }));
    fireEvent.click(screen.getByRole('button', { name: '+' }));
    fireEvent.click(screen.getByRole('button', { name: '5' }));
    fireEvent.click(screen.getByRole('button', { name: '-' }));
    fireEvent.click(screen.getByRole('button', { name: '3' }));
    fireEvent.click(screen.getByRole('button', { name: '=' }));
    expect(screen.getByText('22')).toBeTruthy();
  });
});
