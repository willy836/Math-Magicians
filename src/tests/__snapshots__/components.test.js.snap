// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home and Quote components render correctly Home component should match the snapshot 1`] = `
<div
  className="home"
>
  <h2
    className="home-h2"
  >
    Welcome to our page!
  </h2>
  <div
    className="home-content"
  >
    <p
      className="home-para"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Magni fugiat veniam iure explicabo! Magni iusto fugit, quo illo vero eaque. Recusandae laboriosam deleniti provident sit qui quasi tenetur iure quam rerum maxime ipsum suscipit eligendi quae incidunt sint doloribus harum, illum error iusto natus ipsam! Ad corrupti dolore delectus praesentium.
    </p>
    <p
      className="home-para"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Tenetur omnis odit vero iure reiciendis voluptas molestias numquam sequi? Deserunt voluptas accusantium labore laudantium consequuntur natus repudiandae sit quas quam aliquid fugiat commodi itaque, possimus deleniti eum modi delectus eligendi at quaerat ea cum ullam. Impedit iusto totam dolore pariatur harum doloribus corporis amet minima recusandae odit illo ipsum cum ratione atque, aliquid odio nesciunt consequuntur facere quos? Ea obcaecati neque placeat laborum aperiam, accusamus sunt excepturi veniam, adipisci qui distinctio.
    </p>
  </div>
</div>
`;

exports[`Home and Quote components render correctly Quote component should match the snapshot 1`] = `
<div
  className="random-q"
>
  <p
    className="quote"
  >
    Mathematics is not about numbers, equations, computations, or algorithms: it is about understanding. - William Paul Thurston
  </p>
</div>
`;

exports[`Test links in Header component Header links render correctly 1`] = `null`;
